cmd/agent.go

```diff
@@ -0,0 +1,39 @@
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/pyroscope-io/pyroscope/pkg/cli"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+// agentCmd represents the agent command
+var agentCmd = &cobra.Command{
+	Use:   "agent [flags]",
+	Short: "starts pyroscope agent.",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return cli.StartAgent(&cfg.Agent)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(agentCmd)
+
+	cli.PopulateFlagSet(&cfg.Agent, agentCmd.Flags(), cli.WithSkip("targets"))
+	viper.BindPFlags(agentCmd.Flags())
+
+	// Here you will define your flags and configuration settings.
+
+	// Cobra supports Persistent Flags which will work for this command
+	// and all subcommands, e.g.:
+	// agentCmd.PersistentFlags().String("foo", "", "A help for foo")
+
+	// Cobra supports local flags which will only run when this command
+	// is called directly, e.g.:
+	// agentCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	agentCmd.SetUsageFunc(func(cmd *cobra.Command) error {
+		fmt.Println(gradientBanner() + "\n" + DefaultUsageFunc(cmd.Flags(), cmd))
+		return nil
+	})
+}
```

Comment:
Would it be better to do initialise commands in a more explicit way without global vars and `init`s? Like:
```go
func newAgentCmd(cfg *config.Agent) *cobra.Command
```

Reply:
Again something that cobra generates and something that made me feel a bit uneasy as well, especially with global vars which I always like to avoid. But when I thought about why they would do it that way a bit more I realized it might be okay in this case. These are just configuration variables that are created and initialized in the beginning and never changed again, so having them won't create tight coupling or decrease readability or even hinder our ability to test these things. So I'm not sure whether I should abandon cobra "recommendation" in this case unless there are some downfalls I'm not seeing?